File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/TouristSpotModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TouristSpotModel {
    private Long id;

    private String name;
    private String spotType;
    private String location;
    private Double entranceFee;
    private Integer difficultyLevel;
    private String restrictions;

    private List<PackageSpotModel> packages;
}

File: frontend/tour-design-web/src/app/models/tourist-spot.model.ts
// src/app/models/tourist-spot.model.ts
import { PackageSpotModel } from './package-spot.model';

export interface TouristSpotModel {
  id: number;
  name: string;
  spotType?: string;
  location?: string;
  entranceFee?: number;
  difficultyLevel?: number;
  restrictions?: string;

  packages?: PackageSpotModel[];
}
